test/fail_compilation/fail5153.d

```diff
@@ -0,0 +1,27 @@
+// https://issues.dlang.org/show_bug.cgi?id=5153
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail5153.d(25): Error: cannot initialize a `struct` object using `new`. Remove the `new` keyword.
```

Comment:
It may be an overkill, but what about something along the lines of:
```
The type of `new Foo` is `Foo*` (pointer to `Foo`), whereas the typeof the variable `f` is `Foo`. Consider removing the `new` keyword before the constructor call, or change the type of `f` to `Foo*`.
```

Reply:
`lhs` of the AssignExp may not necessarily be a variable declaration, it might be just `f` or `some.object.f`. In order to put the above mentioned message, which I certainly think is an overkill, I would have to specifically test for this particular case (where `lhs` is a declaration expression). I don't think that that level of specificity is required and from my experience when a user uses `Foo f = new Foo(0)` in the vast majority of cases the fix is `Foo f = Foo(0)` and not `Foo *f = new Foo(0)`